src/controllers/confirmation.controller.ts

```diff
@@ -1,13 +1,25 @@
 import { NextFunction, Request, Response } from "express";
 import { Templates } from "../types/template.paths";
 import { urlUtils } from "../utils/url";
+import { REVIEW_PATH } from "../types/page.urls";
+import { logger } from "../utils/logger";
 
 export const get = (req: Request, res: Response, next: NextFunction) => {
   try {
-    const transactionId = urlUtils.getTransactionIdFromRequestParams(req);
-    return res.render(Templates.CONFIRMATION, {
-      referenceNumber: transactionId,
-    });
+    const paymentStatus = req.query.status;
+    if (paymentStatus === "paid") {
+      const transactionId = urlUtils.getTransactionIdFromRequestParams(req);
+      return res.render(Templates.CONFIRMATION, {
+        referenceNumber: transactionId,
+      });
+    } else {
+      logger.info("Payment status: " + paymentStatus + " - redirecting to the review page");
+      const companyNumber = urlUtils.getCompanyNumberFromRequestParams(req);
```

Comment:
there is a shortcut way of building the url by using urlUtils.getUrlToPath and pass in the req object. Should reduce the number of lines then.